extensions/blocks/podcast-episodes/edit.js

```diff
@@ -3,49 +3,129 @@
 /**
  * External dependencies
  */
+import { Component } from '@wordpress/element';
+import {
+	Button,
+	Disabled,
+	ExternalLink,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	TextControl,
+	Toolbar,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { BlockIcon } from '@wordpress/block-editor';
-import { Placeholder, withNotices } from '@wordpress/components';
-import { useState } from '@wordpress/element';
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
 
 /**
  * Internal dependencies
  */
 import './editor.scss';
-import icon from './icon';
-
-function PodcastEpisodesEdit( {
```

Comment:
why we are re-writing the component using `class`? There is some strong reason to refactor in this sense?
AFAIK, inline functions seem to be a very good trend when creating new components.

Reply:
Agreed. That said, I wonder whether we ought to respect the existing JP style guide?

@jeherve are React hook based functional components allowed in Jetpack?

Reply:
You'll find both approaches in the codebase, but we've been moving towards functional components in recent blocks to follow Gutenberg's direction in that regard.

Reply:
I took a look and found quickly three blocks that are using inline functions:
[Amazon](https://github.com/Automattic/jetpack/blob/master/extensions/blocks/amazon/edit.js#L35-L51), [Calendly](https://github.com/Automattic/jetpack/blob/master/extensions/blocks/calendly/edit.js#L38-L47), [ContactInfo](https://github.com/Automattic/jetpack/blob/master/extensions/blocks/contact-info/edit.js#L36), ... Pretty sure there are more

Reply:
Let's do this in a new PR then yes?

Reply:
It's already done in this one, Dave

Reply:
oh, the commit that I did was discarded. b06ff4572bd4887c83f0f4cc11c453e1d0460bdb

Reply:
Let's go with the Funcitonal approach. @retrofox is going to introduce this shortly.

Reply:
✅ https://github.com/Automattic/jetpack/pull/14948/commits/318e7203c36e3a24102eec814e970708d5cbafa7